src/GraphQl/Resolver/Factory/CollectionResolverFactory.php

```diff
@@ -84,17 +94,38 @@ public function __invoke(string $resourceClass = null, string $rootClass = null,
             if (isset($rootClass, $source[$rootProperty = $info->fieldName], $source[ItemNormalizer::ITEM_KEY])) {
                 $rootResolvedFields = $this->identifiersExtractor->getIdentifiersFromItem(unserialize($source[ItemNormalizer::ITEM_KEY]));
                 $subresource = $this->getSubresource($rootClass, $rootResolvedFields, array_keys($rootResolvedFields), $rootProperty, $resourceClass, true, $dataProviderContext);
+
+                if (null !== $this->dispatcher) {
+                    $this->dispatcher->dispatch(PreReadEvent::NAME, new PreReadEvent($subresource));
```

Comment:
I don't think this is a pre-read here since the subresource has been already fetched.